src/middleware.ts

```diff
@@ -1,23 +1,30 @@
-import { withAuth } from 'next-auth/middleware';
+import { NextRequestWithAuth } from 'next-auth/middleware';
 import { NextResponse } from 'next/server';
+import { handleMobileAuth, nextAuthMiddleware } from './lib/middleware-utils';
 
-export const config = {
-  matcher: ['/courses/:path*', '/questions/:path*', '/bookmarks'],
-};
+const PUBLIC_ROUTES = [
+  '/',
+  '/privacy-policy',
+  '/refund',
+  '/tnc',
+  '/api/auth/mobile/login',
+  '/signin',
+];
 
-export default withAuth(async (req) => {
-  if (process.env.LOCAL_CMS_PROVIDER) return;
+export async function middleware(request: NextRequestWithAuth) {
+  const pathname = request.nextUrl.pathname;
 
-  const token = req.nextauth.token;
-  if (!token) {
-    return NextResponse.redirect(new URL('/invalidsession', req.url));
+  // Check public routes
+  if (PUBLIC_ROUTES.includes(pathname)) {
+    return NextResponse.next();
   }
-  const user = await fetch(
-    `${process.env.NEXT_PUBLIC_BASE_URL_LOCAL}/api/user?token=${token.jwtToken}`,
-  );
 
-  const json = await user.json();
-  if (!json.user) {
-    return NextResponse.redirect(new URL('/invalidsession', req.url));
-  }
-});
+  // Mobile Auth
+  await handleMobileAuth(request);
```

Comment:
can we here get also a device type? like
deviceType:"mobile"...

so we could do
```
if(deviceType==='mobile'){
await handleMobileAuth(request);
return NextResponse.next();
}
return (nextAuthMiddleware as any)(request);
```

something like this??

Reply:
Will have to use useragent, can add it later, if needed.

Don't think we need this urgently I guess

Reply:
@nimit9 for now there will be only desktop people, and for each login request we are unnecessary calling mobileAuth  
It is also really simple to implement it, probably 2 minutes

Reply:
We'll be checking headers for user agent, and in handleMobileAuth also we're checking headers for bearer token.

But yeah will be easy to implement as you said. Happy to do it if you say so

Reply:
added the check @siinghd

Reply:
<3